LDK/resources/web/LDK/panel/TabbedReportPanel.js

```diff
@@ -1067,12 +1072,16 @@ Ext4.define('LDK.panel.TabbedReportPanel', {
 
             for (var i=0;i<token.length;i++){
                 var t = token[i].split(':');
+                t[0] = decodeURIComponent(t[0]);
+                if (t.length > 1) {
+                    t[1] = decodeURIComponent(t[1]);
+                }
                 switch(t[0]){
                     case 'inputType':
                         context.inputType = t[1];
                         break;
                     case 'showReport':
-                        this.isReportTabSelected = (t[1] === 1);
+                        this.isReportTabSelected = (t[1] === '1');
                         break;
                     case 'activeReport':
                         var report = decodeURI(t[1]);
```

Comment:
On review, I think this at least needs discussion, since it's going to be a behavior change.

As you know, decodeURIComponent and decodeURI behave differently.  Previously, we did not decode special characters in the activeReport param  (, / ? : @ & = + $ #).  With the original code, special characters for t[1] were not decoded. Now they are, on line 1077.  the value of t[1] is redundantly decoded on 1087, when decodeURI() is called. I cannot currently think of a reason that's a bug, but want to point it out.

In EHR usage, activeReport is the name defined in the database (ehr.reports), which probably doesnt have a lot of control/validation over it. I doubt names have special characters, but I guess it's possible.  In laboratory module, the report name is more programmatic, like: 

activeReport:org.labkey.laboratory.ExtraDataSourcesDataProvider||Laboratory||ExtraDataSources||tabReport||General||project_usage||Cohorts/Groups

The browser doesnt appear to bother encoding ':'. I believe both cases will still work whether decoding uses decodeURIComponent  or decodeURI.

Anyway, I propose one of the following:

- on 1077, use decodeURI to be conservative and remove the redundant call on 1087.
- on 1077 keep decodeURIComponent and remove the redundant call of decodeURI on 1087

Reply:
For background:

https://www.labkey.org/ONPRC/Support%20Tickets/issues-details.view?issueId=40945

which was a followup from this issue:

https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=36231

Since t is a String array, t[1] will never === 1, so prior to this change, the code was effectively:

this.isReportTabSelected = false;

but given your comments, you may not be worried about that part of the change. You point on the double-decode is a good one.

Since these are components of a URI and not a full URI, I feel like we should be consistently using encodeURIComponent and decodeURIComponent, so I'm inclined to go with your second proposal. I'll commit in a moment.